### ISSUE TITLE
Web Components as an alternative to frameworks

### ISSUE THREAD

**Issue by dragontheory** [kamranahmedse/developer-roadmap#888]
#### What roadmap is this issue about?

- [x] Frontend Roadmap
- [ ] Backend Roadmap
- [ ] DevOps Roadmap
- [ ] All Roadmaps

#### What is this issue about?

- [ ] Discussion for a pull request I would want to open.
- [ ] Addition of a new item
- [ ] Removal of some existing item
- [x] Changing in arrangement
- [x] General suggestion
- [x] Sharing an Idea
- [ ] Something else

#### Please acknowledge the below listed

- [x] This is not a duplicate issue. I have searched and there is no existing issue for this.
- [x] I understand that these roadmaps are highly opinionated. The purpose is to not to include everything out there in these roadmaps but to have everything that is most relevant today comparing to the other options listed.
- [x] I have read the [contribution docs](../contributing.md) before opening this issue.

#### Enter the details about the issue here

Thank you for your time and effort in publishing these roadmaps. They are very helpful! 

If I may, I would like to suggest that Web Components could be listed along with the frameworks. Or even put on the path before the frameworks. 

The reason being, Web Components are now the WHATWG/W3C standard and supported in all major browsers. As the standard, Web Components are what the companies who make the browsers strive to enable, with or without a framework. Frameworks come and go. With Web Components, the browser itself IS the framework, if you will, but without the overhead, dependencies, and learning curve of a framework. Plus, since Web Components are compatible with any framework, front-end career oriented developers still receive the knowledge/understanding/skills fundamental for other frameworks. 

For all the reasons, I have been hearing/reading/seeing that more and more companies are opting for the vanilla Web Component route.  

Forgive me if I am preaching to the choir. 

Thanks again for your hard work. 

Have a great day!

**Comment by pdurbin** [comment]
I completely agree. It would be fantastic to have Web Components as a first class option under "Skill Based" next to React, Angular, Vue, etc.

Since this issue was opened, Web Components was at least added to https://roadmap.sh/frontend with contributions by @kamranahmedse @aroyan and @GuilloSGit in PRs such as these:

- #1412
- #1413
- #1414
- #1587

Here's how it looks right now:

<img width="653" alt="Screen Shot 2022-10-16 at 8 46 38 AM" src="https://user-images.githubusercontent.com/21006/196036582-783bb001-929e-4197-b8de-7729f2a17582.png">

(By the way, I heard about roadmap.sh on [20MinJS](https://20minjs.com/) [episode 32](https://podcasts.apple.com/us/podcast/episode-32-switching-careers-to-web-development-with/id1613196630?i=1000582235304). Great project!)
